src/app/questions-container/questions-container.component.ts

```diff
@@ -0,0 +1,116 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { HttpClientModule, HttpClient } from '@angular/common/http';
+import { CommonModule } from '@angular/common';
+import { HeaderComponent } from '../header/header.component';
+import { ScorePageComponent } from '../score-page/score-page.component';
+
+interface Question {
+  question: string;
+  options: string[];
+  answer: string;
+}
+
+interface Quiz {
+  title: string;
+  icon: string;
+  questions: Question[];
+}
+
+interface QuizData {
+  quizzes: Quiz[];
+}
+
+@Component({
+  selector: 'app-questions-container',
+  standalone: true,
+  imports: [
+    CommonModule,
+    HeaderComponent,
+    HttpClientModule,
+    ScorePageComponent,
+  ],
+  templateUrl: './questions-container.component.html',
+  styleUrls: ['./questions-container.component.css'],
+})
+export class QuestionsContainerComponent implements OnInit {
+  @Input() public subject = '';
+  public quizzes: Quiz[] = [];
+  @Input() public selectedQuiz: Quiz = {
```

Comment:
```suggestion
  @Input() selectedQuiz!: Quiz
```
and don't access modifiers to the input it is not really necessary as well as the angualr default functions/interfaces e.g onInit()